flujo/validation/linters.py

```
@@ -0,0 +1,1522 @@
+                    if not is_mergeable and _PR is not None:
+                        try:
+                            is_mergeable = isinstance(pout, type) and issubclass(pout, _PR)
+                        except Exception:
+                            is_mergeable = False
+                    if is_mergeable:
+                        continue
+
+                    # Escalation when next step does not consume prev output (rough heuristic)
+                    def _consumes_prev() -> bool:
+                        try:
+                            in_t = getattr(step, "__step_input_type__", Any)
```

Ruff (0.12.2)

Function definition does not bind loop variable `step`

(B023)